suave/release-checklist.md

```diff
@@ -0,0 +1,36 @@
+suave-geth Release Checklist
+============================
+
+Note: the best days to release a new version are Monday to Wednesday. Never release on a Friday (because of the risk of needing to work on the weekend).
+
+### Prepare the release
+
+- [ ] Run linter and tests:
+  ```bash
+  $ make lint && make test && make suave
+  $ ./build/bin/suave-geth version
+  ```
+- [ ] Test latest version with [suapp-examples](https://github.com/flashbots/suapp-examples) and [suave-std](https://github.com/flashbots/suave-std)
+
+### Update documentation (if needed)
+
+Prepare documentation updates before the release is published. Only if needed.
+
+- [ ] [Docs](https://github.com/flashbots/suave-docs)
+- [ ] [Specs](https://github.com/flashbots/suave-specs)
+
+### Publish the release
+
+- [ ] Pick the version (i.e. `v0.2.0`)
+- [ ] Tag new version (`git tag -s v0.2.0`)
+- [ ] Push tag to Github, CI will create a draft release
+- [ ] Edit the draft release to prepare nice release notes
+- [ ] Publish the release (note: this will send an email to subscribers on Github)
+
+### Test the published release
+
+- [ ] Download the release binary: `curl -L https://suaveup.flashbots.net | bash` and check the version with `suave-geth version`
```

Comment:
TODO: This does not show the version of suave-geth but the version from the geth fork.

Reply:
hehe, yeah that's a good point! definitely need to add this, as it's critical information for checking and debugging

Reply:
it's defined here: https://github.com/flashbots/suave-geth/blob/release-checklist/params/version.go#L23-L29

i'm assuming we'll just override this version number with the suave-geth one?

Reply:
added to publish steps

Reply:
updated

Reply:
```bash
./build/bin/suave-geth version

Suave-Geth
Version: 0.1.1-dev
Git Commit: bbb9d5ee4f6fba64e31231092858484303aa82d0
Git Commit Date: 20240110
Architecture: arm64
Go Version: go1.21.0
Operating System: darwin
GOPATH=/Users/metachris/go
GOROOT=
```